src/js/controller/model.js

```diff
@@ -94,11 +94,24 @@ define([
                     this.set('buffer', data.bufferPercent);
                 /* falls through */
                 case events.JWPLAYER_MEDIA_META:
+                    var metaStreamType = utils.streamType(data.duration, data.seekableRange, this.get('minDvrWindow'));
+                    this.set('streamType', metaStreamType);
+                    this.set('seekableRange', data.seekableRange);
+
                     var duration = data.duration;
                     if (_.isNumber(duration) && !_.isNaN(duration)) {
                         mediaModel.set('duration', duration);
                         this.set('duration', duration);
                     }
+
+                    // For legacy purposes, report duration as negative video duration when in DVR mode
+                    evt = {
+                        duration: metaStreamType === 'DVR' ? -data.videoDuration : data.duration,
+                        height: data.height,
```

Comment:
Now all providers need to output `videoDuration` and `seekableRange` for DVR functionality to work.

Should there be a short circuit for legacy providers that return negative duration? Or, documentation for impact to existing providers.